pages/exercises/index.js

```diff
@@ -0,0 +1,63 @@
+import { exercises as initialExercises } from "@/lib/exercises";
+import { useState } from "react";
+import {
+  StyledFlexWrapper,
+  ExerciseList,
+  StyledFilterButton,
+  StyledFilterWrapper,
+} from "@/styledComponents";
+import Exercise from "@/components/Exercise/Exercise";
+import FilterSection from "@/components/FilterSection/FilterSection";
+
+export default function Exercises() {
+  const [exercises, setExercises] = useState(initialExercises);
+  const [filters, setFilters] = useState([]);
+  const [filterMode, setFilterMode] = useState(false);
+
+  function filterExercises(exercises, filters) {
+    const filteredArray = exercises.filter((exercise) => {
+      return filters.every((filter) => exercise.muscleGroups.includes(filter));
+    });
+    return filteredArray;
+  }
+
+  function onAddFilter(newFilter) {
+    setFilters([...filters, newFilter]);
+  }
+
+  function handleDisableFilter(deleteFilter) {
+    const newFiltersArray = filters.filter((filter) => filter !== deleteFilter);
+    setFilters(newFiltersArray);
+  }
+
+  function handleClear() {
+    setFilters([]);
+    setExercises(initialExercises);
+  }
+
+  return (
+    <StyledFlexWrapper>
+      <ExerciseList>
+        <StyledFilterWrapper>
+          <StyledFilterButton onClick={() => setFilterMode(!filterMode)}>
+            {filterMode ? "Filter ▲" : "Filter ▼"}
+          </StyledFilterButton>
+          {filterMode ? (
+            <FilterSection
+              filters={filters}
+              onAddFilter={onAddFilter}
+              handleDisableFilter={handleDisableFilter}
+              handleClear={handleClear}
+            />
+          ) : null}
+        </StyledFilterWrapper>
```

Comment:
I suggest to place the filter outside of the list as all direct children from a list should be list items (`li` elements). Whenever you are unsure, you can check it here: https://caninclude.glitch.me/